File: 0897-40-2206-divide-array-into-equal-pairs/0897-40-2206-divide-array-into-equal-pairs.java
class Solution {
    public boolean divideArray(int[] nums) {
        int xr = 0;
        int xr2 = 0;
        for(int i = 0 ; i < nums.length;i++){
            xr^=nums[i];
            xr2^=(nums[i]+1);
        }
        return xr == 0 && xr2==0;
    }
}

File: 0897-40-2206-divide-array-into-equal-pairs/0897-40-2206-divide-array-into-equal-pairs.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var divideArray = function(nums) {
    let xr = 0,xr2=0;
    for(let x of nums){
        xr^=x;
        xr2^=(x+1);
    }
    return xr == 0 && xr2==0;
};